### PR TITLE
PW-4930 Add requiredShopperFields field to CreatePaymentLinkRequest

### PR DESCRIPTION
**Pull request by saquibsayyad**
**Description**
Added requiredShopperFields field in  CreatePaymentLinkRequest

**Tested scenarios**
Create payment link flow.

### CODE REVIEW

File: src/main/java/com/adyen/model/checkout/InstallmentOption.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
+/*
```
Comment (Aleffio) [inline src/main/java/com/adyen/model/checkout/InstallmentOption.java:1]: very minor: could you please add the correct copyright and auto-indent the file from intellij?

File: src/main/java/com/adyen/model/checkout/CreatePaymentLinkRequest.java [hunk 2]
```diff
@@ -133,7 +130,7 @@ public void write(final JsonWriter jsonWriter, final RecurringProcessingModelEnu
 
             @Override
             public RecurringProcessingModelEnum read(final JsonReader jsonReader) throws IOException {
-                String value = jsonReader.nextString();
+                Object value = jsonReader.nextString();
```
Comment (rkewlani) [inline src/main/java/com/adyen/model/checkout/CreatePaymentLinkRequest.java:133]: possible null pointer. why it is changed to Object from String?

File: src/main/java/com/adyen/model/checkout/InstallmentOption.java (hunk 1 above)
Comment (saquibsayyad) [reply src/main/java/com/adyen/model/checkout/InstallmentOption.java:1]: will do it, thanks.

File: src/main/java/com/adyen/model/checkout/CreatePaymentLinkRequest.java (hunk 2 above)
Comment (saquibsayyad) [reply src/main/java/com/adyen/model/checkout/CreatePaymentLinkRequest.java:133]: seems it was changed in the model generator, i will revert it back.
Regarding NPE for `JsonReader`, I don't think we need to handle it as it is called by GSON internally.

**Review by Aleffio — APPROVED** [review]

**Review by rkewlani — APPROVED** [review]

**Pull request merged**